docs(chatbot): add component pages (#244)
siemens/ix-docs: docs/components/ai-message/guide.md

---
doc-type: 'tab-item'
description: 'Usage guide for AI message'
---

# AI message - Usage

> AI messages display a single assistant response inside a conversational thread. We recommend using them for answers users need to read, review and act on.

Reviewer: ```suggestion
AI messages display individual responses from the AI (agent, bot, Copilot, etc.) inside a conversational thread once the prompt input field has been completed and sent by the user.  
```
Question: is the component for the prompt input field the same as the message component?

![AI message anatomy](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=7959-520&t=8Rj3ErabF16Vm3lH-4)

1. Content
2. Actions

## Options

> - **Content:** Keep the response easy to scan. We usually structure longer answers with a short lead sentence followed by lists, short paragraphs or other lightweight content blocks.

Reviewer: This is recommended but we have no control over the format of the response. Instead here you should say what the font type and font size is. And the font color as we can control this. If we have this info and if suitable here. If you want to give recommendations about AI behavior feedback you can refer to: https://ix.siemens.io/docs/guidelines/conversational-design/designing-conversations/overview

> - **Actions:** Add only message-level actions that users expect after reading, e.g. copy, rate response quality or regenerate. We recommend using subtle tertiary [icon buttons](../icon-button) so actions stay available without competing with the answer.

Reviewer: Question: Is regenerate usually within this component? 

Please refer to the AI terminology Figma file for wording with action icons: https://www.figma.com/design/lqjt9c5IzzwQ4eJ4nqG7Kv/AI-Terminology?node-id=1-9&t=d5UkOPKJfj9qDmYM-1 ONLY INTERNAL

Reviewer 2: Yes, following the AI guidelines. Is there something off?

> - **Sources:** If the response is grounded in files, web results or internal data, expose that provenance close to the message content, e.g. with [chips](../chip). Only show sources when they are real and inspectable.

Reviewer: There are two kinds of resources - inline and also under the message box. Do we want to talk about both? I think both use chips. Need to check. 

Also note "Only show sources when they are real and inspectable." - this must be removed as it is not possible to control. 

```suggestion
- **Sources:** Display all sources after the AI message response within the message box with [chips](../chip). 
```

Reviewer 2: Here it's only about the ones under the message box since it is an actual option to enable.

Sources get inspectable by simply adding them to the list - if there is no list, the sources chip makes no sense to show. 
For the "real"-part we can discuss 😂 even though I think the bot is only references that we (app builders) feed them, so eventually we're in charge of available sources, arent' we? 🤔  

For inline usage there's currently no official component in place, since the official (internal) guideline suggested a chip with less than 24px height (-> hit area too small).

Reviewer 2: Decision: find a better wording for inspectable

## Behavior in context

> - **Responsiveness:** AI messages use from 45 to 80 % of the chat's container width, depending on the viewport width.

Reviewer: ```suggestion
- **Responsiveness:** AI messages use from 45 to 80% of the chat's container width, depending on the viewport width.
```

## Dos and Don’ts

> - Do structure longer responses into short paragraphs or lists so users can scan them quickly

Reviewer: remove - cant have this do
```suggestion
```

Reviewer 2: why not?

> - Do show clear [loading indicators](../spinner/) while the assistant is processing

Reviewer: please link to https://ix.siemens.io/docs/guidelines/conversational-design/essentials/wording-terms

> - Do show clear [loading indicators](../spinner/) while the assistant is processing

Reviewer: ```suggestion
- Do show clear [loading indicators](../spinner/) while the assistant is generating responses. 
```

> - Do show sources or evidence when the response refers to external or retrieved information

Reviewer: ```suggestion
- Do use chips to show sources
```

Reviewer 2: If you think this recommendation is not needed, I would remove it altogether since we have the chips-rule already in the options above.

> - Don't overuse actions by focusing only on the few actions users commonly need after reading, e.g. copy, feedback, regenerate

Reviewer: maybe would make this a do: 

"Do use the same actions for each AI message for consistency, but no more than 4 to avoid overloading users"

Reviewer 2: love that one much better <3

> - Don’t present speculative or low-confidence output as if it were verified fact

Reviewer: ```suggestion
- Don’t add thumbs up or down actions unless you have aligned with data protection guidelines
```

Reviewer 2: I'll write that as a Do to avoid double negation: "Do add thumbs up or down actions only if you have aligned with data protection guidelines"

Could you just explain (for my personal understanding) how thumbs up/down connects to data protection?

---
doc-type: 'tab-item'
description: 'Usage guide for AI message'
---

# AI message - Usage

> AI messages display a single assistant response inside a conversational thread. We recommend using them for answers users need to read, review and act on, not for transient status updates that fit better in [toasts](../toast) or [progress indicators](../progress-indicator).

Reviewer 3: Surprising to me to differentiate the ai message from toast and progress indicator, these would not be components I'd expect from an AI response. For me, it would be sufficient to stop the sentence after "act on."

![AI message anatomy](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=7959-520&t=8Rj3ErabF16Vm3lH-4)

1. Content
2. Actions

## Options

- **Content:** Keep the response easy to scan. We usually structure longer answers with a short lead sentence followed by lists, short paragraphs or other lightweight content blocks.
- **Actions:** Add only message-level actions that users expect after reading, e.g. copy, rate response quality or regenerate. We recommend using subtle tertiary [icon buttons](../icon-button) so actions stay available without competing with the answer.
- **Sources:** If the response is grounded in files, web results or internal data, expose that provenance close to the message content, e.g. with [chips](../chip). Only show sources when they are real and inspectable.

## Behavior in context

- **Responsiveness:** AI messages use 45 to 80 % of the chat container width.

## Dos and Don’ts

- Do structure longer responses into short paragraphs or lists so users can scan them quickly
- Do show clear [loading indicators](../spinner/) while the assistant is processing

> - Do offer only the few actions users commonly need after reading, e.g. copy, feedback or regenerate

Reviewer 3: to keep dos and don'ts balanced, this would qualify to be formulated as a don't to not overuse actions
